pkg/user/tx_client.go

```diff
@@ -253,10 +268,12 @@ func (client *TxClient) BroadcastPayForBlobWithAccount(ctx context.Context, acco
 
 // SubmitTx forms a transaction from the provided messages, signs it, and submits it to the chain. TxOptions
 // may be provided to set the fee and gas limit.
-func (client *TxClient) SubmitTx(ctx context.Context, msgs []sdktypes.Msg, opts ...TxOption) (*sdktypes.TxResponse, error) {
+func (client *TxClient) SubmitTx(ctx context.Context, msgs []sdktypes.Msg, opts ...TxOption) (*TxResponse, error) {
 	resp, err := client.BroadcastTx(ctx, msgs, opts...)
-	if err != nil {
-		return resp, err
+	if err != nil && resp != nil {
+		return &TxResponse{Code: resp.Code, TxHash: resp.TxHash}, fmt.Errorf("failed to broadcast tx: %v", err)
+	} else if err != nil {
+		return &TxResponse{}, fmt.Errorf("failed to broadcast tx: %v", err)
 	}
```

Comment:
nit: would be a little cleaner to use a helper function:
```
if err != nil {
    return parseTxResp(resp), fmt.Errorf("failed to broadcast tx: %w", err)
}
```
which turns the broadcastTxResponse to a TxResponse or nil if it's nil

Reply:
yeah i considered it but I thought that maybe this was more straight-forward and easy to understand